library/core/functions.render.php

```diff
@@ -1734,6 +1734,10 @@ function signInUrl($target = '', $force = false) {
             }
         }
 
+        if (strpos($target, 'entry') === 0) {
```

Comment:
This could probably be tightened up a little bit to force a match on the entry controller, instead of any controller (or path) that starts with the word "entry" (e.g. "entry-exam/p1").

```suggestion
        if (strpos($target, 'entry/') === 0) {
```

Maybe add another test case for that too.